src/modules/SourceDialogs.rb

```diff
@@ -2713,6 +2713,19 @@ def valid_scheme?(url)
     publish :function => :TypePopup, :type => "string ()"
     publish :function => :TypeDialog, :type => "symbol ()"
     publish :function => :TypeDialogDownloadOpt, :type => "map <string, any> ()"
+
+  private
+
+    # Build URL from tokens
+    # @param [Hash] with keys:"scheme","http","host","port","path","query",...
+    # @return [String] url, empty string if invalid data is used to build the url.
+    def build_url(url_settings)
+      # URL.Build is escaping characters (e.g. ":") which is wrong for
+      # pathes like /dist/ibs/SUSE:/SLE-12-SP1:/GA. (bnc#966413)
+      # So we are unescaping it.
+      URLRecode.UnEscape(URL.Build(url_settings))
```

Comment:
in fact it looks like bug in URL.Build as according to its source https://github.com/yast/yast-yast2/blob/5762181d62762816a73fc040362c1efb5d97deed/library/types/src/modules/URL.rb#L91 `:` should be escaped only for user and password part of URL, not for path which is your example.